qemu/tests/fio_perf.py

```diff
@@ -0,0 +1,259 @@
+    result_file = open(result_path, "w")
+
+    # scratch host and windows guest version info
+    get_version(session, result_file, guest_ver_cmd, os_type,
+                driver_format, cmd_timeout)
+
+    # online disk
+    if os_type == "windows":
+        for num in range(1, 2):
+            disks = check_status(session, cmd_timeout, num)
+            diskstatus = re.findall("Disk\s+\d+\s+(\w+).*?\s+\d+", disks[0])[0]
+            if diskstatus == "Offline":
+                online_disk_cmd = params.get("online_disk_cmd")
+                online_disk_run = online_disk_cmd % num
+                (s, o) = session.cmd_status_output(online_disk_run,
+                                                   timeout=cmd_timeout)
+                if s:
+                    raise exceptions.TestFail("Failed to online disk: %s" % o)
+    # format disk
+    if format == "True":
+        session.cmd(pre_cmd, cmd_timeout)
+
+    # get order_list
+    order_line = ""
+    for order in order_list.split():
+        order_line += "%s|" % format_result(order)
+
+    # get result tested by each seanario
+    for io_pattern in rw.split():
+        result_file.write("Category:%s\n" % io_pattern)
+        result_file.write("%s\n" % order_line.rstrip("|"))
+        if io_pattern == "rewrite":
+            io_pattern = "write"
+        for bs in block_size.split():
+            for io_depth in iodepth.split():
+                for numjobs in threads.split():
+                    line = ""
+                    line += "%s|" % format_result(bs[:-1])
+                    line += "%s|" % format_result(io_depth)
+                    line += "%s|" % format_result(numjobs)
+                    if format == "True":
+                        file_name = io_pattern + "_" + bs + "_" + io_depth
+                        run_cmd = fio_cmd % (io_pattern, bs, io_depth, file_name, numjobs)
+                    else:
+                        run_cmd = fio_cmd % (io_pattern, bs, io_depth, numjobs)
+
+                    logging.info("run_cmd is: %s" % run_cmd)
+                    if os_type == "linux":
+                        (s, o) = session.cmd_status_output(drop_cache,
+                                                           timeout=cmd_timeout)
+                        if s:
+                            raise exceptions.TestFail("Failed to free memory: %s" % o)
+                    io_exits_b = int(process.system_output("cat /sys/kernel/debug/kvm/exits"))
+                    fio_t = threading.Thread(target=fio_thread, args=(session, run_cmd, cmd_timeout))
+                    cpu_t = threading.Thread(target=cpu_thread, args=("mpstat 1 60", "/tmp/cpus"))
+                    fio_t.start()
+                    cpu_t.start()
+                    fio_t.join()
+                    cpu_t.join()
+
+                    io_exits_a = int(process.system_output("cat /sys/kernel/debug/kvm/exits"))
+                    vm.copy_files_from(params.get("guest_result_file"), "/tmp/")
```

Comment:
Please move all temp files to the path getting by `virttest.data_dir.get_tmp_dir()`.